test/Microsoft.AspNet.Mvc.Xml.Test/XmlAssertTest.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.AspNet.Mvc.Xml
+{
+    public class XmlAssertTest
+    {
+        [Theory]
+        [InlineData("<A>hello</A>", "<A>hey</A>")]
+        [InlineData("<A>hello</A>", "<A>hey</A>")]
+        [InlineData("<A><B>hello world</B></A>", "<A><B>hello world!!</B></A>")]
+        [InlineData("<a><b>hello</b><b>world</b></a>", "<a><b>hello</b><c>world</c></a>")]
+        [InlineData("<a><b>hello</b><b>world</b></a>", "<a><b>hello</b><b attribute=\"value\">world</b></a>")]
+        public void ValidatesTextNodes(string input1, string input2)
+        {
+            var equalException = Assert.Throws<EqualException>(() => XmlAssert.Equal(input1, input2));
+            Assert.Equal(input1, equalException.Expected);
+            Assert.Equal(input2, equalException.Actual);
+        }
+
+        [Theory]
+        [InlineData("<A></A>", "<A></A>")]
+        [InlineData("<A>hello<B>world</B>hello</A>", "<A>hello<B>world</B>hello</A>")]
+        [InlineData("<A><!-- comment1 --><B></B></A>", "<A><!-- comment1 --><B></B></A>")]
+        [InlineData("<A/>", "<A/>")]
+        [InlineData("<A><![CDATA[<greeting></greeting>]]></A>", "<A><![CDATA[<greeting></greeting>]]></A>")]
+        [InlineData("<a>hello<b>world</b>hello</a>", "<a>hello<b>world</b>hello</a>")]
+        [InlineData(
+            "<a>hello<b color=\"red\" siz=\"medium\">world</b>hello</a>",
+            "<a>hello<b siz=\"medium\" color=\"red\">world</b>hello</a>")]
+        public void DoesNotFail_OnEmptyTextNodes(string input1, string input2)
```

Comment:
rename to `Validates_WithEmptyElements` to avoid mentioning an impossible case; XML parsers don't create empty text nodes